### PR TITLE
upgrade .net framework to v4.7.2

### PR DESCRIPTION
**Pull request by ohine**
## Summary
This pull request updates the framework for all projects to v4.7.2.

### CODE REVIEW

File: DNN Platform/DotNetNuke.Web/Common/DotNetNukeHttpApplication.cs [hunk 1]
```diff
@@ -115,7 +115,7 @@ private void Application_Start(object sender, EventArgs eventArgs)
             DotNetNukeShutdownOverload.InitializeFcnSettings();
 
             // register the assembly-lookup to correct the breaking rename in DNN 9.2
-            DotNetNuke.Services.Zip.SharpZipLibRedirect.RegisterSharpZipLibRedirect();
+           // DotNetNuke.Services.Zip.SharpZipLibRedirect.RegisterSharpZipLibRedirect();
```
Comment (bdukes) [inline DNN Platform/DotNetNuke.Web/Common/DotNetNukeHttpApplication.cs]: Is this change supposed to be part of this PR?

File: DNN Platform/Tests/DotNetNuke.Tests.Integration/packages.config [hunk 2]
```diff
@@ -5,8 +5,8 @@
   <package id="Moq" version="4.2.1502.0911" targetFramework="net45" allowedVersions="[4.2.1502.0911]" />
   <package id="NBuilder" version="5.0.0" targetFramework="net45" allowedVersions="[5.0.0]" />
   <package id="Newtonsoft.Json" version="10.0.3" targetFramework="net45" allowedVersions="[10.0.3]" />
-  <package id="NSubstitute" version="3.1.0" targetFramework="net45" allowedVersions="[3.1.0]" />
+  <package id="NSubstitute" version="3.1.0" targetFramework="net45" allowedVersions="[3.1.0]" requireReinstallation="true" />
```
Comment (bdukes) [inline DNN Platform/Tests/DotNetNuke.Tests.Integration/packages.config]: looks like these packages need to be [reinstalled](https://docs.microsoft.com/en-us/nuget/consume-packages/reinstalling-and-updating-packages)

File: DNN Platform/DotNetNuke.Web/Common/DotNetNukeHttpApplication.cs (hunk 1 above)
Comment (ohine) [reply DNN Platform/DotNetNuke.Web/Common/DotNetNukeHttpApplication.cs]: No, good catch! I'll undo that. I thought I reset things before but I guess I must of started from a different branch. Oppps :(

File: DNN Platform/Tests/DotNetNuke.Tests.Integration/packages.config (hunk 2 above)
Comment (ohine) [reply DNN Platform/Tests/DotNetNuke.Tests.Integration/packages.config]: Ok, I'll get that cleaned up as well. Thanks @bdukes for the review!

**Comment by ZenSoftware** [comment]
Should all of the `packages.config` be updated to use the [Target Framework Moniker](https://docs.microsoft.com/en-us/dotnet/standard/frameworks) of `targetFramework="net472"` as well?

**Comment by bdukes** [comment]
@ZenSoftware, I _think_ that for the packages where it matters, they've been marked as `requireReinstallation`, otherwise there isn't a difference between the `net45` and `net472` versions. I think we would want to keep them at `net45` so that NuGet knows that the package was installed using an older framework (so if the package starts to differentiate between the two versions in the future, NuGet knows that a reinstallation is required)

**Comment by mitchelsellers** [comment]
@ohine Did you want to update this to resolve the conflicts?  Or would you prefer that I make a new PR?

File: DNN Platform/Admin Modules/Dnn.Modules.Console/Dnn.Modules.Console.csproj [hunk 3]
```diff
@@ -13,13 +13,15 @@
...
-    <TargetFrameworkVersion>v4.5</TargetFrameworkVersion>
+    <TargetFrameworkVersion>v4.7.2</TargetFrameworkVersion>
     <UseIISExpress>true</UseIISExpress>
     <IISExpressSSLPort />
     <IISExpressAnonymousAuthentication />
     <IISExpressWindowsAuthentication />
     <IISExpressUseClassicPipelineMode />
     <UseGlobalApplicationHostFile />
+    <TargetFrameworkProfile />
+    <Use64BitIISExpress />
```
Comment (ahoefling) [inline DNN Platform/Admin Modules/Dnn.Modules.Console/Dnn.Modules.Console.csproj]: Not sure if you intended to add this to the .csproj

File: DNN Platform/Admin Modules/Dnn.Modules.Console/Dnn.Modules.Console.csproj [hunk 4]
```diff
@@ -48,22 +50,21 @@
...
     </Reference>
     <Reference Include="Microsoft.CSharp" />
+    <Reference Include="System.Data.DataSetExtensions" />
     <Reference Include="System.Web.DynamicData" />
     <Reference Include="System.Web.Entity" />
     <Reference Include="System.Web.ApplicationServices" />
     <Reference Include="System.ComponentModel.DataAnnotations" />
     <Reference Include="System" />
     <Reference Include="System.Data" />
-    <Reference Include="System.Core" />
```
Comment (ahoefling) [inline DNN Platform/Admin Modules/Dnn.Modules.Console/Dnn.Modules.Console.csproj]: System.Core is removed, was that intentional?

File: DNN Platform/Admin Modules/Dnn.Modules.ModuleCreator/Dnn.Modules.ModuleCreator.csproj [hunk 5]
```diff
@@ -13,12 +13,15 @@
...
-    <TargetFrameworkVersion>v4.5</TargetFrameworkVersion>
+    <TargetFrameworkVersion>v4.7.2</TargetFrameworkVersion>
     <UseIISExpress>true</UseIISExpress>
     <IISExpressSSLPort />
     <IISExpressAnonymousAuthentication />
     <IISExpressWindowsAuthentication />
     <IISExpressUseClassicPipelineMode />
+    <TargetFrameworkProfile />
+    <Use64BitIISExpress />
+    <UseGlobalApplicationHostFile />
```
Comment (ahoefling) [inline DNN Platform/Admin Modules/Dnn.Modules.ModuleCreator/Dnn.Modules.ModuleCreator.csproj]: Not sure if you intended to add this to the .csproj

File: DNN Platform/Admin Modules/Dnn.Modules.ModuleCreator/Dnn.Modules.ModuleCreator.csproj [hunk 6]
```diff
@@ -47,22 +50,21 @@
...
     </Reference>
     <Reference Include="Microsoft.CSharp" />
+    <Reference Include="System.Data.DataSetExtensions" />
     <Reference Include="System.Web.DynamicData" />
     <Reference Include="System.Web.Entity" />
     <Reference Include="System.Web.ApplicationServices" />
     <Reference Include="System.ComponentModel.DataAnnotations" />
     <Reference Include="System" />
     <Reference Include="System.Data" />
-    <Reference Include="System.Core" />
```
Comment (ahoefling) [inline DNN Platform/Admin Modules/Dnn.Modules.ModuleCreator/Dnn.Modules.ModuleCreator.csproj]: Did you intend to remove System.Core?

File: DNN Platform/Modules/CoreMessaging/DotNetNuke.Modules.CoreMessaging.csproj [hunk 7]
```diff
@@ -73,6 +75,10 @@
     <Reference Include="System.Runtime.Serialization" />
     <Reference Include="System.Web" />
     <Reference Include="System.Web.Abstractions" />
+    <Reference Include="System.Web.ApplicationServices" />
+    <Reference Include="System.Web.DynamicData" />
+    <Reference Include="System.Web.Entity" />
+    <Reference Include="System.Web.Extensions" />
```
Comment (ahoefling) [inline DNN Platform/Modules/CoreMessaging/DotNetNuke.Modules.CoreMessaging.csproj]: These all look like newly added assemblies, did you mean to add them?

**Comment by ahoefling** [comment]
@ohine I started to look through the changes and I am seeing some artifacts that got added. I am not sure if they are intentional but we should clarify them. 

* Some assemblies you remove `System.Core`, was this intentional? Do we not need it?
* Several assemblies we are adding new csproj elements that may cause issues (not 100% sure)
    * `Use64BitIISExpress`
    * `TargetFrameworkProfile`
    * `UseGlobalApplicationHostFile`
* nit: There are several projects that reordered assembly references, which makes it difficult to review the changes
* nit: the tools version was updated to 12 which is fine, but we should be careful about updating that above 14 as that could start causing breaking changes for developers using older versions of visual studio

I understand my comments are very particular but I don't want to see artifacts introduced. Overall this is great work on getting us to where we need to be with .NET Framework and then .NET Standard

**Comment by ahoefling** [comment]
@ohine if you need help with this or want to pair, let me know. I would love to get this in to one of the next 9.x releases so we can start working towards .NET Standard initiatives we need for Razor Pages.

**Comment by ohine** [comment]
Unfortunately, I deleted my reforked the repo a few months ago because my branches were completely messed up. Which, unfortunately deleted the branch that this work was based in. It looks like we will have to redo it. If @ahoefling or @mitchelsellers or any of the others would like to handle this, that would be excellent. I'm currently focused on testing the RC for the next RC.

**Comment by ohine** [comment]
I will also add, the majority of the changes were all done via the VS upgrade process. All I did was update the framework to v4.6x and then to v4.7x. The reason I made a stop at v4.6x was at the time the group was still unsure which version we'd be jumping too.

**Comment by ahoefling** [comment]
Ill start working on a PR for this, thanks for the update

**Pull request closed without merging**